x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/checkpoint/DefaultCheckpointProvider.java

```diff
@@ -311,4 +322,34 @@ public void getCheckpointingInfo(long lastCheckpointNumber, IndexerState nextChe
             getIndexCheckpoints(checkpointsByIndexListener);
         }
     }
+
+    /**
+     * Inspect source changes and report differences
+     *
+     * @param lastSourceIndexes the set of indexes seen in the previous checkpoint
+     * @param newSourceIndexes the set of indexes seen in the new checkpoint
+     */
+    void reportSourceIndexChanges(final Set<String> lastSourceIndexes, final Set<String> newSourceIndexes) {
+        // spam protection: only warn the first time
+        if (newSourceIndexes.isEmpty() && lastSourceIndexes.isEmpty() == false) {
+            String message = "Source did not resolve to any concrete indexes";
```

Comment:
```suggestion
            String message = "Source did not resolve to any open indexes.";
```
I am not 100% sure we should use the terms "concrete". That could be considered an internal term/idea